week-3-practise/level-2/2-map.js

```diff
@@ -14,7 +14,17 @@ var attendances = [
   ["Nina", 10]
 ];
 
-var eligibleStudentNames; // TODO: Complete this line.
+function eligibleForExam(attendance) {
+  if (attendance[1] >= 8) {
+    return attendance[0];
+  } else {
+    return "";
+  }
```

Comment:
When you write a function for use with `filter` or `find`, it should return `true` or `false` for each input, not a string. Don't try to rely on Javascript's odd behaviour. The filter and map are two separate things so shouldn't share the same function. It's better to have something like:

```
var eligibleStudentNames = attendances
  .filter(isEligibleStudent)
  .map(studentName)
```